ImageViewer/Source/UtilityFunctions.swift

```diff
@@ -0,0 +1,116 @@
+    
+    let horizontalOffest = (boundingSize.width > contentSize.width) ? ((boundingSize.width - contentSize.width) * 0.5): 0.0
+    let verticalOffset = (boundingSize.height > contentSize.height) ? ((boundingSize.height - contentSize.height) * 0.5): 0.0
+    
+    return CGPoint(x: contentSize.width * 0.5 + horizontalOffest,  y: contentSize.height * 0.5 + verticalOffset)
+}
+
+func zoomRect(ForScrollView scrollView: UIScrollView, scale: CGFloat, center: CGPoint) -> CGRect {
+    
+    let width = scrollView.frame.size.width  / scale
+    let height = scrollView.frame.size.height / scale
+    let originX = center.x - (width / 2.0)
+    let originY = center.y - (height / 2.0)
+    
+    return CGRect(x: originX, y: originY, width: width, height: height)
+}
+
+func screenshotFromView(view: UIView) -> UIImage {
+    
+    let image: UIImage
+    
+    UIGraphicsBeginImageContextWithOptions(view.bounds.size, true, UIScreen.mainScreen().scale)
+    view.drawViewHierarchyInRect(view.bounds, afterScreenUpdates: false)
+    image = UIGraphicsGetImageFromCurrentImageContext()
+    UIGraphicsEndImageContext()
+    
+    return image
+}
+
+func rotationTransform() -> CGAffineTransform {
+    
+    return CGAffineTransformMakeRotation(degreesToRadians(rotationAngleToMatchDeviceOrientation(UIDevice.currentDevice().orientation)))
+}
+
+func degreesToRadians(degree: CGFloat) -> CGFloat {
+    return CGFloat(M_PI) * degree / 180
+}
+
+private func rotationAngleToMatchDeviceOrientation(orientation: UIDeviceOrientation) -> CGFloat {
+    
+    var desiredRotationAngle: CGFloat = 0
+    
+    switch orientation {
+    case .LandscapeLeft:                    desiredRotationAngle = 90
+    case .LandscapeRight:                   desiredRotationAngle = -90
+    case .PortraitUpsideDown:               desiredRotationAngle = 180
+    default:                                desiredRotationAngle = 0
+    }
+    
+    return desiredRotationAngle
+}
+
+func rotationAdjustedBounds() -> CGRect {
+    
+    let applicationWindow = UIApplication.sharedApplication().delegate?.window?.flatMap { $0 }
+    guard let window = applicationWindow else { return UIScreen.mainScreen().bounds }
+    
+    if isPortraitOnly() {
+        
+        return (UIDevice.currentDevice().orientation.isLandscape) ? CGRect(origin: CGPointZero, size: window.bounds.size.inverted()): window.bounds
+    }
+    
+    return window.bounds
+}
+
+func isPortraitOnly() -> Bool {
+    
+    return UIApplication.sharedApplication().supportedInterfaceOrientationsForWindow(nil) == .Portrait
+}
+
+func maximumZoomScale(forBoundingSize boundingSize: CGSize, contentSize: CGSize) -> CGFloat {
+    
+    //we want to allow the image to always cover 4x the area of screen
+    return min(boundingSize.width, boundingSize.height) / min(contentSize.width, contentSize.height) * 4
+}
+
+func rotationAdjustedCenter(view: UIView) -> CGPoint {
+    guard isPortraitOnly() else {
+        return view.center
+    }
```

Comment:
Here the return goes to the next line, but on the `rotationAdjustedBounds` method it stays in the same line. Be consistent.